src/olm/account/mod.rs

```diff
@@ -439,6 +439,45 @@ impl Account {
 
         pickle.try_into()
     }
+
+    /// Create a dehydrated device pickle.
+    pub fn to_dehydrated_device(&self, key: &[u8; 32]) -> Result<String, crate::LibolmPickleError> {
+        use self::dehydrated_device::{expand_pickle_key, Pickle};
+        use crate::utilities::pickle_libolm;
+
+        let pickle_key = expand_pickle_key(key, &self.curve25519_key().to_base64());
+
+        pickle_libolm::<Pickle>(self.into(), pickle_key.as_ref())
+    }
+
+    /// Create an [`Account`] object by rehydrating a device.
+    pub fn from_dehydrated_device(
+        pickle: &str,
+        curve25519key: &str,
+        key: &[u8; 32],
+    ) -> Result<Self, crate::LibolmPickleError> {
+        use self::dehydrated_device::{expand_pickle_key, Pickle, PICKLE_VERSION};
+        use crate::utilities::unpickle_libolm;
+
+        let pickle_key = expand_pickle_key(key, curve25519key);
+
+        #[cfg(feature = "libolm-compat")]
+        return unpickle_libolm::<Pickle, _>(pickle, pickle_key.as_ref(), PICKLE_VERSION).or_else(
```

Comment:
I had tried that, but the compiler doesn't like it because it thinks it isn't a final statement (even though this `#[cfg]` and the next one are mutually exclusive).

Reply:
Since I'm going to change the encryption, it probably makes sense to deal with compatibility in the SDK layer rather than here, so the `#[cfg]` will go away.